### ISSUE TITLE
🐛 BUG: Wrangler Publish breaks Jenkins builds

### ISSUE THREAD

**Issue by iTechTonyChristopher** [cloudflare/wrangler2#1832]
### What version of `Wrangler` are you using?

2.1.1

### What operating system are you using?

node:gallium-alpine3.14

### Describe the Bug

Our Jenkins Pipeline uses wrangler on a docker image to publish our Cloudflare workers. This latest build has broken our pipelines when you make a manual change. It is prompting the user to overwrite the manual changes, which breaks our pipeline:
```
 ⛅️ wrangler 2.1.1 
-------------------
▲ [WARNING] You are about to publish a Workers Service that was last published via the Cloudflare Dashboard.

  		Edits that have been made via the dashboard will be overridden by your local code and config.


Would you like to continue? (y/n)
[GWould you like to continue? (y/n)
Error: Raw mode is not supported on the current process.stdin, which Ink uses as input stream by default.
```

Steps to reproduce:

1. Create a worker by code and publish to a git repository
2. Create a Docker file to install wrangler and run wrangler publish
3. Create a Jenkins Pipeline to build and execute the Docker file
4. Manually change the published worker using Quick edit feature in Cloudflare Dashboard
5. Make a change to the worker in code and push to git repo
6. Run Jenkins Pipeline again - This will fail due to the prompt.

Temporary solution: Use wrangler 2.0.29

**Comment by petebacondarwin** [comment]
We should have a command line option in `wrangler publish` to force the deployment even if the Worker was changed in the dashboard.
But this is actually a valid warning to you @iTechTonyChristopher - if you have made changes in the dashboard to a Worker then deploying it again via Jenkins and Wrangler will overwrite those changes.

**Comment by threepointone** [comment]
Let's not have the dialog block when it's not interactive.

**Comment by petebacondarwin** [comment]
Not showing a warning/dialog could potentially hide a deployment overwriting an important change.
I think we should require the user to opt-in rather than doing it by default. We are trying to discourage users from using both Wrangler and Dashboard to modify Workers.

**Comment by threepointone** [comment]
You could show the warning, but no interactive dialogs should present themselves in stuff like CI. That's the six for this issue at the moment. 

Regarding the broader point of not encouraging and making it an error, it's a breaking change and should be rolled out accordingly, probably in a major.

**Comment by petebacondarwin** [comment]
> Regarding the broader point of not encouraging and making it an error, it's a breaking change and should be rolled out accordingly, probably in a major.

Fair point on the breaking change.

**Comment by iTechTonyChristopher** [comment]
> We should have a command line option in `wrangler publish` to force the deployment even if the Worker was changed in the dashboard. But this is actually a valid warning to you @iTechTonyChristopher - if you have made changes in the dashboard to a Worker then deploying it again via Jenkins and Wrangler will overwrite those changes.

Fair point but this is expected behaviour as it's how it acts in 2.0.29. I completely understand adding in warnings like this being useful for people but it needs better documenting and an option to automatically say yes for pipelines like this. I knew my changes would be overwritten, I tested them by manually updating and then update my codebase and push. And yes, we know it's not the best way, getting Wrangler Dev working is in our backlog.

**Comment by rozenmd** [comment]
This should be fixed in wrangler@2.1.3 @iTechTonyChristopher

**Comment by iTechTonyChristopher** [comment]
Amazing, really fast response. Thank you.
